src/main/java/com/cs525/reversi/services/GameServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package com.cs525.reversi.services;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.cs525.reversi.models.Game;
+import com.cs525.reversi.models.GameStatus;
+import com.cs525.reversi.models.MatrixRow;
+import com.cs525.reversi.models.User;
+import com.cs525.reversi.repositories.GameRepository;
+import com.cs525.reversi.repositories.UserRepository;
+import com.cs525.reversi.req.NewGame;
+
+@Service
+public class GameServiceImpl implements GameService {
+
+	@Autowired
+	private GameRepository gameRepo;
+	@Autowired
+	private UserRepository userRepo;
+
+	@Override
+	public String createNewGame(NewGame newGameForm) {
+
+		User player2 = userRepo.findByUsername(newGameForm.getUserName());
+		User player1 = userRepo.findByUsername("comp");
+
+		if (player2 == null) {
+			player2 = new User();
```

Comment:
@MustafaMosad Lombok's `@Data` annotation creates a constructor for `@NonNull` fields... so you can use that to construct and set username and uuid in one go

Reply:
never mind we are using the `@NoArgumentConstructor` annotation which reverses that behavior